edumanage/models.py

```diff
@@ -100,30 +101,71 @@ def value_to_string(self, obj):
         value = self._get_val_from_obj(obj)
         return self.get_db_prep_value(value)
 
-ERTYPES = (
-    (1, 'IdP only'),
-    (2, 'SP only'),
-    (3, 'IdP and SP'),
+# https://www.djangosnippets.org/snippets/2402/
+def get_namedtuple_choices(*choices_tuples):
```

Comment:
Hi @zmousm ,

I think it would be useful to document the required structure of the choices_tuples passed to this call.

From the code here, I can each tuple in the choices should start with an identifier, but I see the structure then quite varies:
* ER_TYPES has a numeric ID, others don't.
* ER_TYPES has has a short handle *after* a descriptive name, others reverse this order.

I think it would be good to define the structure to be passed - and then adhere to it.

Reply:
Hello @vladimir-mencl-eresearch,

the structure is the sequence of iterables that [Django accepts for choices](https://docs.djangoproject.com/en/1.11/ref/models/fields/#choices), and in each of them the two items are preceded by the [field name for namedtuple](https://docs.python.org/3/library/collections.html#collections.namedtuple). _ERTYPES is special in that each iterable also includes at the end the string value, which is new in eduroam db v2, so as to help with the mapping. However any such further items are ignored by `get_namedtuple_choices`.

Coming up, I will be adding some documentation.

Reply:
Agree that when I treat `ER_TYPES` as a separate special case, the structures look more uniform - a tuple of 3-tuples, each starting with an identifier.

Still, sometimes, the second field in each tuple (the internal value to store?) is a string, sometimes a number.  I assume the last field in each tuple is a description.

Having this documented would help a lot - thanks a lot in advance for that.

Reply:
> Still, sometimes, the second field in each tuple (the internal value to store?) is a string, sometimes a number. I assume the last field in each tuple is a description.

That is correct, per the Django spec.

Reply:
Thanks for the patience with me - happy with just pointing to Django doc.